python/ray/data/dataset.py

```diff
@@ -4464,11 +4464,24 @@ def types(self) -> List[Union[Literal[object], "pyarrow.DataType"]]:
     def __eq__(self, other):
         return isinstance(other, Schema) and other.base_schema == self.base_schema
 
-    def __str__(self):
-        return f"Schema({dict(zip(self.names, self.types))})"
-
```

Comment:
If `__str__` isn't implemented, it falls back to `__repr__`. We don't need to implement both.